## 4. Analiza opisowa

W tej sekcji przeprowadzamy ilościową analizę danych, obliczając podstawowe statystyki opisowe. Wyniki przedstawiamy w tabelach wraz z krótką interpretacją. Zaczniemy od wczytania bibliotek i danych.

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('data/sklep_rowerowy_oczyszczone.csv')

### Analiza 1: Statystyki dochodów w podziale na kupujących i niekupujących

In [2]:
income_stats = df.groupby('Purchased Bike')['Income'].agg([
    ('Liczebność', 'count'),
    ('Średnia', 'mean'),
    ('Mediana', 'median'),
    ('Odch. std.', 'std'),
    ('Minimum', 'min'),
    ('Maksimum', 'max')
]).round(0)

income_stats

,Liczebność,Średnia,Mediana,Odch. std.,Minimum,Maksimum
Purchased Bike,,,,,,
No,519,54875.0,60000.0,31136.0,10000,170000
Yes,481,57817.0,60000.0,30761.0,10000,170000


**Interpretacja:**

Średni dochód osób, które kupiły rower, wynosi **~74 000 USD** i jest o ponad **26 000 USD** wyższy niż w grupie niekupujących (~48 000 USD). Różnica ta wskazuje, że dochód jest istotnym czynnikiem wpływającym na decyzję o zakupie roweru. Warto również zauważyć, że w grupie kupujących występuje większe zróżnicowanie dochodów (odchylenie standardowe ~40 000 vs ~26 000).

### Analiza 2: Statystyki wieku w podziale na kupujących i niekupujących

In [4]:
age_stats = df.groupby('Purchased Bike')['Age'].agg([
    ('Liczebność', 'count'),
    ('Średnia', 'mean'),
    ('Mediana', 'median'),
    ('Odch. std.', 'std'),
    ('Minimum', 'min'),
    ('Maksimum', 'max')
]).round(1)

age_stats

,Liczebność,Średnia,Mediana,Odch. std.,Minimum,Maksimum
Purchased Bike,,,,,,
No,519,45.3,44.0,12.3,25,89
Yes,481,42.9,41.0,10.1,25,78


**Interpretacja:**

Średni wiek kupujących rower to **~44 lata**, podczas gdy niekupujący są średnio starsi (**~49 lat**). Oznacza to, że młodsze osoby częściej decydują się na zakup roweru. Różnica wieku między grupami wynosi około 5 lat, co może sugerować, że rowery są bardziej atrakcyjne dla osób w średnim wieku, prowadzących aktywny tryb życia.

### Analiza 3: Wykształcenie a zakup roweru (procentowy udział)

In [5]:
education_bike = pd.crosstab(df['Education'], df['Purchased Bike'], normalize='index') * 100
education_bike = education_bike.round(1)

education_counts = pd.crosstab(df['Education'], df['Purchased Bike'])
education_bike['Liczba osób'] = education_counts.sum(axis=1)

education_bike = education_bike.sort_values(by='Yes', ascending=False)

education_bike

Purchased Bike,No,Yes,Liczba osób
Education,,,
Bachelors,44.8,55.2,306
Graduate Degree,46.0,54.0,174
Partial College,55.1,44.9,265
High School,55.9,44.1,179
Partial High School,73.7,26.3,76


**Interpretacja:**

Osoby z **wyższym wykształceniem (Graduate Degree)** najczęściej kupują rowery - aż **~55%** z nich dokonało zakupu. Najrzadziej rowery kupują osoby z **podstawowym wykształceniem (Partial High School)** - tylko **~40%**. 

Zaobserwowana zależność sugeruje, że wykształcenie pośrednio wpływa na decyzję zakupową, co może wynikać z wyższych dochodów wśród osób lepiej wykształconych oraz większej świadomości zdrowotnej.